### ISSUE TITLE
The custom webp file extension is not being excluded.

### ISSUE THREAD

**Issue by hgati** [baldwin-agency/magento2-module-image-cleanup#3]
I am saving images with a .ngx.webp file extension as dynamic images. 
I have named nginx-related webp images as .ngx.webp in order to control webp images in nginx. 
This is because there are also manually uploaded files with the .webp extension, and I need to distinguish them so that I can later delete only the nginx-related webp files.
So, I have configured it as follows to exclude ngx.webp, but it doesn't seem to be excluded.

![](https://gcdnb.pbrd.co/images/dmg4QKoZjRQu.png)

### The Result.

![](https://gcdnb.pbrd.co/images/Z03GKu4f9qCG.png)

**Comment by hostep** [comment]
This probably happens because instead of replacing the `.jpg` file extension with `.ngx.webp`, it gets appended, so it becomes `.jpg.ngx.webp`.

Do you have the opportunity to remove `.jpg` from the filename? I have the suspicion that the module won't delete those files anymore after doing so.

Alternatively, we could try to change [this piece of code](https://github.com/baldwin-agency/magento2-module-image-cleanup/blob/509d3d8b02c1e6076be5cd45f0f42cb412e3cadd/Finder/UnusedFilesFinder.php#L211-L215) from:
```php
            foreach ($alternativeExtensions as $ext) {
                $extraFiles[] = $fileWithoutExtension . '.' . $ext;
            }
```
to:
```php
            foreach ($alternativeExtensions as $ext) {
                $extraFiles[] = $fileWithoutExtension . '.' . $ext;
                $extraFiles[] = $filename . '.' . $ext;
            }
```

That might also solve your problem.

Would you be able to find some time to test this?

**Comment by hgati** [comment]
_It gets appended as follows._

![](https://gcdnb.pbrd.co/images/30qv63RucxcP.png)

**Comment by hostep** [comment]
Okay, so this change fixes it in your case? It no longer tries to delete those images now?

I don't think there are downsides to this fix. So if you can confirm, I'll add this fix in a new version a bit later this week.

**Comment by hgati** [comment]
> Okay, so this change fixes it in your case? It no longer tries to delete those images now?
> 
> I don't think there are downsides to this fix. So if you can confirm, I'll add this fix in a new version a bit later this week.

I will back up over 7,000 images from the live server to the local development server, test them, and let you know the results.

**Comment by hgati** [comment]
Okay, It has been deleted as expected.

**Continue with the deletion of these [y/N]?**

![](https://gcdnb.pbrd.co/images/KF9XPHqsexir.png)

**Cleaned up 18 files, was able to cleanup**

![](https://gcdnb.pbrd.co/images/TYKnmFvf4BWa.png)

**Comment by hostep** [comment]
Fix for this was released in [version 1.2.2](https://github.com/baldwin-agency/magento2-module-image-cleanup/releases/tag/v1.2.2).

Thanks for reporting this case and confirming the fix!
